src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -224,22 +220,13 @@ private void OnHostingHttpRequestInStop(object arg)
                 return;
             }
 
-            IScope scope = tracer.ActiveScope;
+            var scope = tracer.ActiveScope;
```

Comment:
Note to reviewers: `ISpan` was not letting us use `Span.SetServerStatusCode()` extension method. `scope` is now of type `Span`.

Reply:
Scope is now Span?

Reply:
Oops, thanks. I meant `Scope` (instead of `IScope`). Fixed my comment above.

Reply:
`IScope.Span` is of type `ISpan`.
`Scope.Span` is of type `Span`, which is what we need.

Reply:
Yep, very similar to opentracing structure. I wonder if it worth to change the extension method to use ISpan instead, maybe not.

Reply:
Separate PR  :P